vm/src/stdlib/os.rs

```diff
@@ -273,6 +273,35 @@ fn get_right_permission(
     }
 }
 
+#[cfg(target_os = "macos")]
+fn getgroups() -> nix::Result<Vec<Gid>> {
+    use libc::{c_int, gid_t};
+    use std::ptr;
+    let ret = unsafe { libc::getgroups(0, ptr::null_mut()) };
+    let mut groups = Vec::<Gid>::with_capacity(Errno::result(ret)? as usize);
+    loop {
+        let ret = unsafe {
+            libc::getgroups(
+                groups.capacity() as c_int,
+                groups.as_mut_ptr() as *mut gid_t,
+            )
+        };
+
+        match Errno::result(ret) {
```

Comment:
Errno::result(err).map should also work here

Reply:
I am not familiar with transformed it into the map.
Can you guide it, please?

Reply:
You could do something like `return Errno::result(ret).map(|s| { unsafe { set_len... }; groups});`, because you're not transforming the err case at all, so there's no need to do that whole `match`.

Reply:
You could do something like `return Errno::result(ret).map(|s| { unsafe { set_len... }; groups});`, because you're not transforming the err case at all, so there's no need to do that whole `match`.

Reply:
@coolreader18 
Thanks alot!